src/contracts/core/AllocationManager.sol

```diff
@@ -195,25 +211,48 @@ contract AllocationManager is
+
             // if there is a pending deallocation, slash pending deallocation proportionally
+            // since the delta is negative, we increase it to make sure they can free less of it when completed
             if (mInfo.pendingMagnitudeDelta < 0) {
-                int128 slashedPending =
-                    int128(uint128(uint256(uint128(-mInfo.pendingMagnitudeDelta)).mulWad(wadToSlash)));
-                mInfo.pendingMagnitudeDelta += int128(slashedPending);
+                uint64 pendingMagnitudeDeltaSlashed =
+                    uint64(uint256(uint128(-mInfo.pendingMagnitudeDelta)).mulWad(wadToSlash));
+                mInfo.pendingMagnitudeDelta += int128(uint128(pendingMagnitudeDeltaSlashed));
+                slashedMagnitude += pendingMagnitudeDeltaSlashed;
             }
-            // update operatorMagnitudeInfo
-            _operatorMagnitudeInfo[operator][strategies[i]][operatorSetKey] = mInfo;
 
-            // 3. update totalMagnitude, get total magnitude and subtract slashedMagnitude
+            // update operatorMagnitudeInfo
+            _setMagnitudeInfo({
+                operator: operator,
+                strategy: strategies[i],
+                operatorSetKey: operatorSetKey,
+                mInfo: mInfo,
+                isCompletion: false
+            });
+
+            // 3. update encumberedMagnitude, subtract the slashed magnitude
+            _updateEncumberedMagnitude({
+                operator: operator,
+                strategy: strategies[i],
+                encumberedMagnitudeDelta: -int128(uint128(slashedMagnitude))
+            });
+
+            // 3. update totalMagnitude, subtract the slashed magnitude and push the new totalMagnitude to the history
             Snapshots.DefaultWadHistory storage totalMagnitudes = _totalMagnitudeUpdate[operator][strategies[i]];
-            uint64 totalMagnitudeBeforeSlashing = uint64(totalMagnitudes.latest());
-            totalMagnitudes.push({key: uint32(block.timestamp), value: totalMagnitudeBeforeSlashing - slashedMagnitude});
+            uint64 totalMagnitudeAfterSlash = uint64(totalMagnitudes.latest()) - slashedMagnitude;
+            totalMagnitudes.push({key: uint32(block.timestamp), value: totalMagnitudeAfterSlash});
+            emit TotalMagnitudeUpdated(operator, strategies[i], totalMagnitudeAfterSlash);
 
             // 4. Decrease operators shares in the DM
-            delegation.decreaseOperatorShares(
-                operator, strategies[i], totalMagnitudeBeforeSlashing, uint64(totalMagnitudes.latest())
-            );
+            delegation.decreaseOperatorShares({
+                operator: operator,
+                strategy: strategies[i],
+                previousTotalMagnitude: totalMagnitudeAfterSlash + slashedMagnitude,
+                newTotalMagnitude: totalMagnitudeAfterSlash
+            });
         }
+
+        emit OperatorSlashed(operator, _decodeOperatorSet(operatorSetKey), description);
```

Comment:
I feel like we'll have to change this to make it easier for offchain to associate the slashes  with the associated mag changes. Perhaps something like a slashID per operator??

Reply:
any decrease such that `block.timestamp == effectTimestamp` is a slash

any decrease in the future that overwrites a previous decrease a the same time is a slash.

I can easily add these checks in the `OperatorSetMagnitudeUpdate` and add a bool, would that work?

Reply:
or are you looking at connecting a reason to a slash?

perhaps we don't allow two slashes in the same tx/block for the same operatorSet?

Reply:
How does the subgraph distinguish between two slashes for different slash reasons that occurred in the same block from a multicall?

Reply:
right. shall we add a
```solidity
///  @notice operator => strategy => operator set (encoded) => last timestamp at which the tuple was slashed
mapping(address => mapping(IStrategy => mapping(bytes32 => uint32))) lastSlashedTimestamp;
```
and enforce its nonzero for the current timestamp

(also i hate to say it: `tstore`!)

Reply:
That seems fine for now, not ideal

Reply:
done https://github.com/Layr-Labs/eigenlayer-contracts/pull/806/commits/d94bf6f353efcdf57f00304560967ba9ca2e3220